crates/server/src/store/in_memory/authority.rs

```diff
@@ -1296,6 +1456,148 @@ impl Authority for InMemoryAuthority {
     ) -> Result<Self::Lookup, LookupError> {
         Ok(AuthLookup::default())
     }
+
+    #[cfg(feature = "dnssec")]
+    async fn get_nsec3_records(
+        &self,
+        name: &LowerName,
+        query_type: RecordType,
+        lookup_options: LookupOptions,
+    ) -> Result<Self::Lookup, LookupError> {
+        // Function to compute the hash of a given name.
+        let hash_name = |name: &Name| {
+            let Nsec3Config {
+                hash_algorithm,
+                ref salt,
+                iterations,
+                ..
+            } = self.nsec3_config;
+            hash_algorithm.hash(salt, name, iterations).unwrap()
+        };
+
+        // Compute the hashed owner name from a given name. This is, the hash of the given name,
+        // followed by the zone name.
+        let get_hashed_name = |name: &LowerName| -> LowerName {
+            let hash = hash_name(name.borrow());
+            let label = data_encoding::BASE32_DNSSEC.encode(hash.as_ref());
+            LowerName::new(
+                &Name::new()
+                    .append_label(label)
+                    .unwrap()
```

Comment:
This is the one I'm most concerned about - I think an attacker could send a crafted request (with a super long non-existent RR) to make the hickory server panic.